HDFDataset.py

```diff
@@ -1039,15 +1047,18 @@ def dump_from_dataset(self, dataset, epoch=1, start_seq=0, end_seq=float("inf"),
         progress_bar_with_time(float(i) / num_seqs)
 
     print("Set seq len info...", file=log.v3)
-    hdf_dataset.create_dataset(attr_seqLengths, shape=(num_seqs, 2), dtype="int32")
+    hdf_dataset.create_dataset(attr_seqLengths, shape=(num_seqs, 1+len(data_target_keys)), dtype="int32")
     for i, seq_len in enumerate(seq_lens):
       data_len = seq_len[default_data_input_key]
-      targets_len = seq_len[default_data_target_key]
-      for data_key in data_target_keys:
-        assert seq_len[data_key] == targets_len, "different lengths in multi-target not supported"
-      if targets_len is None:
-        targets_len = data_len
-      hdf_dataset[attr_seqLengths][i] = [data_len, targets_len]
+      if len(data_target_keys) > 0:
+        targets_len = seq_len[default_data_target_key]
+        for data_key in data_target_keys:
+          assert seq_len[data_key] == targets_len, "different lengths in multi-target not supported"
+        if targets_len is None:
+          targets_len = data_len
+        hdf_dataset[attr_seqLengths][i] = [data_len, targets_len]
```

Comment:
This code is wrong now. E.g. if `len(data_target_keys) == 2`, then you must set 3 lengths here.

Reply:
That's interesting. I did not intend to do anything here except enclosing it into an if block. I think something got messed up in a merge. I'll change it back.

Reply:
It's wrong because you changed the seqLength shape (which before was always 2 items).

Reply:
which I thought was the intended behaviour, but nvm. it should be back to the original version now.

Reply:
Actually I think the shape `(num_seqs, 1 + len(data_target_keys))` is more correct. When I wrote that, I thought that the HDFDataset always expected shape[-1] in {1,2}, but only later I noticed that it is also fine for multiple targets.